Regenerate uuid when copying arrays/table items.

ajboni opened the pull request:
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ✔️
| Is backward-compatible?    | ✔️
| Tests pass?   | ❌
| Fixed issues  | comma-separated list of tickets # fixed by the PR, if any
#1129
| Updated README/docs?   | ❌
| Added CHANGELOG entry?   | ❌

Please review!

schmunk42 commented:
Looks like tests are broken. @ajboni Please have a look.

We should also have a test which ensures a new UUID (non-duplicate) when **copying** and also that values do not change while **reordering**. (CC: @germanbisurgi)

ajboni commented:
Thanks, rewrote it with more cases (arrays of strings vs objects) and added some more tests.
It keeps failing though but seems unrelated (the integer test)

schmunk42 commented:
Thank you! Will be handled in #1134
